examples/platforms/utils/soft_source_match_table.c

```diff
@@ -161,45 +161,45 @@ typedef struct srcMatchExtEntry
 
 static sSrcMatchExtEntry srcMatchExtEntry[RADIO_CONFIG_SRC_MATCH_EXT_ENTRY_NUM];
 
-int8_t utilsSoftSrcMatchExtFindEntry(const otExtAddress *aExtAddress)
+int16_t utilsSoftSrcMatchExtFindEntry(const otExtAddress *aExtAddress)
 {
-    int8_t   entry    = -1;
+    int16_t  entry    = -1;
     uint16_t checksum = sPanId;
 
     checksum += (uint16_t)aExtAddress->m8[0] | (uint16_t)(aExtAddress->m8[1] << 8);
     checksum += (uint16_t)aExtAddress->m8[2] | (uint16_t)(aExtAddress->m8[3] << 8);
     checksum += (uint16_t)aExtAddress->m8[4] | (uint16_t)(aExtAddress->m8[5] << 8);
     checksum += (uint16_t)aExtAddress->m8[6] | (uint16_t)(aExtAddress->m8[7] << 8);
 
-    for (uint8_t i = 0; i < RADIO_CONFIG_SRC_MATCH_EXT_ENTRY_NUM; i++)
+    for (uint16_t i = 0; i < RADIO_CONFIG_SRC_MATCH_EXT_ENTRY_NUM; i++)
     {
         if (checksum == srcMatchExtEntry[i].checksum && srcMatchExtEntry[i].allocated)
         {
-            entry = i;
+            entry = (int16_t)i;
             break;
         }
     }
 
     return entry;
 }
 
-static inline int8_t findSrcMatchExtAvailEntry(void)
+static inline int16_t findSrcMatchExtAvailEntry(void)
 {
-    int8_t entry = -1;
+    int16_t entry = -1;
 
-    for (uint8_t i = 0; i < RADIO_CONFIG_SRC_MATCH_EXT_ENTRY_NUM; i++)
+    for (uint16_t i = 0; i < RADIO_CONFIG_SRC_MATCH_EXT_ENTRY_NUM; i++)
     {
         if (!srcMatchExtEntry[i].allocated)
         {
-            entry = i;
+            entry = (int16_t)i;
             break;
         }
     }
 
     return entry;
 }
 
-static inline void addToSrcMatchExtIndirect(uint8_t entry, const otExtAddress *aExtAddress)
+static inline void addToSrcMatchExtIndirect(uint16_t entry, const otExtAddress *aExtAddress)
```

Comment:
remove inline?

Reply:
The [page](https://www.geeksforgeeks.org/inline-functions-cpp/) lists the cases that can't use `inline` keyword. This function doesn't match any item of the lists. I think it is ok to use `inline` in this function.